sqlglot/dialects/dialect.py

```diff
@@ -137,6 +137,13 @@ def get_start_end(token_type: TokenType) -> t.Tuple[t.Optional[str], t.Optional[
         if enum not in ("", "bigquery"):
             klass.generator_class.SELECT_KINDS = ()
 
+        if enum not in ("", "hive", "spark", "spark2"):
+            modifier_transforms = klass.generator_class.AFTER_HAVING_MODIFIER_TRANSFORMS.copy()
```

Comment:
why do you need to copy?

Reply:
Because otherwise you mutate the parent class' dict